### PR TITLE
Use entry_points for plugins

### PR DESCRIPTION
**Pull request by maxnoe**

### CODE REVIEW

File: docs/changes/2101.feature.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+The ``ctapipe-info`` tool now supports printing information about
+the available ``EventSource`` implementations.
```
Comment (nbiederbeck) [inline docs/changes/2101.feature.rst:2]: and reconstructors.

File: docs/changes/2101.api.rst [hunk 2]
```diff
@@ -0,0 +1,4 @@
+ctapipe now uses entry points for plugin discovery. ``EventSource`` implementations 
+no need to advertise a ``ctapipe_io`` entry point, to be discovered by ctapipe.
```
Comment (StFroese) [inline docs/changes/2101.api.rst:2]: now?

Comment (maxnoe) [reply docs/changes/2101.api.rst:2]: thx

**Comment by maxnoe** [comment]
Ok, the test failure is due to an interplay of pytest-xdist and caplog, I can reproduce the failure locally when running on more than one core, using only pytest the test is passing.

**Review by nbiederbeck — APPROVED** [review]

**Comment by maxnoe** [comment]
@kosack Tests are passing now, please review

**Review by StFroese — APPROVED** [review]

**Review by kosack — APPROVED** [review]

**Pull request merged**